WcaOnRails/app/models/competition.rb

```diff
@@ -1734,7 +1735,7 @@ def set_wcif_events!(wcif_events, current_user)
 
     # Create missing events.
     wcif_events.each do |wcif_event|
-      event_found = competition_events.find_by_event_id(wcif_event["id"])
+      event_found = competition_events.find { |ce| ce.event_id == wcif_event["id"] }
```

Comment:
We call `self.competition_events` above which loads the whole list into memory. By using `#find` we can do lookup in that list, whereas `find_by_event_id` would **always** make a query.

Same below.